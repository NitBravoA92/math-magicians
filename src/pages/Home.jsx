import Layout from './Layout';
import '../styles/Home.css';

const Home = () => (
  <Layout>
    <section id="home">
      <h1 className="page-title">Welcome to our page!</h1>
      <div className="page-content">
        <p className="home-info">
          Math Magicians is a web application aimed at lovers of mathematics and calculators.
          We can say that calculations are a fundamental part of our life.
          We experience them in every aspect of our lives; children, adults, or in our work.
          However, some calculations are easy but there may be some that are somewhat more
          complicated for us and we need the help of a tool that quickly solves them for us.
          That&apos;s where the beloved calculators come in.
        </p>
        <p className="home-info">
          Calculators are a fundamental part of our lives and most of us have to
          use them at one point or another.
          Those people who are highly dependent on calculators can reduce
          their dependency by switching to using an online
          calculator such as the one provided by this website.
        </p>
      </div>
    </section>
  </Layout>
);

export default Home;
