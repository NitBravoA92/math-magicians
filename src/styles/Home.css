.home-info {
  margin-bottom: 2rem;
}
