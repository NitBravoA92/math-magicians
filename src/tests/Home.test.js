import React from "react";
import '@testing-library/jest-dom/extend-expect';
import { render } from "@testing-library/react";
import renderer from 'react-test-renderer';
import Home from "../pages/Home";

describe("The Home component", () => {

  test("renders correctly into the DOM", () => {
    const component = renderer
    .create(<Home />)
    .toJSON();
    expect(component).toMatchSnapshot();
  });

  test("should render the text: Welcome to our page!", () => {
    const component = render(<Home />);

    expect(component.container).toHaveTextContent("Welcome to our page!");
  });
  
  test("should render 2 paragraphs elements", () => {
    const component = render(<Home />);

    const paragraphs = component.container.querySelectorAll("p");
    
    expect(paragraphs.length).toBe(2);
  });

});
