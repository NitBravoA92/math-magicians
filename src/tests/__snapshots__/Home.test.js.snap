// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Home component renders correctly into the DOM 1`] = `
Array [
  <header>
    <a
      className="logo-container"
      href="/"
    >
      <img
        alt="logo"
        className="App-logo"
        src="mathLogo.gif"
      />
      <span
        className="logo-title"
      >
        Math Magicians
      </span>
    </a>
    <nav>
      <ul
        id="menu"
      >
        <li
          className="nav-item"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            href="/calculator"
          >
            Calculator
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            href="/quote"
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main>
    <section
      id="home"
    >
      <h1
        className="page-title"
      >
        Welcome to our page!
      </h1>
      <div
        className="page-content"
      >
        <p
          className="home-info"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. In iure amet architecto hic vel quia repellat recusandae nulla, tempora quo ea velit pariatur autem necessitatibus eius consequuntur porro quae delectus. Lorem ipsum dolor sit amet consectetur adipisicing elit. In iure amet architecto hic vel quia repellat recusandae nulla, tempora quo ea velit pariatur autem necessitatibus eius consequuntur porro quae delectus. Lorem ipsum dolor sit amet consectetur adipisicing elit. In iure amet architecto hic vel quia repellat recusandae nulla, tempora quo ea velit pariatur autem necessitatibus eius consequuntur porro quae delectus. Lorem ipsum dolor sit amet consectetur adipisicing elit. In iure amet architecto hic vel quia repellat recusandae nulla, tempora quo ea velit pariatur autem necessitatibus eius consequuntur porro quae delectus.
        </p>
        <p
          className="home-info"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. In iure amet architecto hic vel quia repellat recusandae nulla, tempora quo ea velit pariatur autem necessitatibus eius consequuntur porro quae delectus. Lorem ipsum dolor sit amet consectetur adipisicing elit. In iure amet architecto hic vel quia repellat recusandae nulla, tempora quo ea velit pariatur autem necessitatibus eius consequuntur porro quae delectus. Lorem ipsum dolor sit amet consectetur adipisicing elit. In iure amet architecto hic vel quia repellat recusandae nulla, tempora quo ea velit pariatur autem necessitatibus eius consequuntur porro quae delectus. Lorem ipsum dolor sit amet consectetur adipisicing elit. In iure amet architecto hic vel quia repellat recusandae nulla, tempora quo ea velit pariatur autem necessitatibus eius consequuntur porro quae delectus.
        </p>
      </div>
    </section>
  </main>,
]
`;
